db.go

```diff
@@ -244,6 +241,12 @@ type HtlcInfo struct {
 func (d *Db) RecordHtlcResolution(ctx context.Context,
 	htlc *HtlcInfo) error {
 
+	// If the database is configured to not store any records, save the hassle of
+	// writing and deleting a record by returning early.
+	if d.fwdHistoryLimit == 0 {
```

Comment:
If the limit is set to zero later, I think you'd want to run `limitHTLCRecords` at least once to truncate the table?

Reply:
👍  added a commit with a once off delete + test